### ISSUE TITLE
[CLOSED] Array wrapper hierarchy generation

### ISSUE THREAD

**Issue by aionbot** [aionnetwork/AVM#73]
**Issue created by [JunhanHu-aion](https://github.com/JunhanHu-aion)** (on Monday Jun 04, 2018 at 17:20 GMT)

Class A{}
Class B extends A{}

B[] instanceof A[] --------> True
B[][][] instanceof A[][][] --------> True

To achieve this behaviour, our generated wrappers need to have the same hierarchy.

**Comment by aionbot** [comment]
**Comment by [jeff-aion](https://github.com/jeff-aion)** (on Tuesday Aug 28, 2018 at 19:38 GMT)

The core of the required work for this item was already done under a8370e6eb7e82305be0424503de20274c4e60dd7, in order to get blake2b working:  the implementation of `Class#avm_desiredAssertionStatus()`.

My work for this item will just involve creating a test to demonstrate how this works and adding some documentation to the `avm_desiredAssertionStatus()` implementation.

**Comment by aionbot** [comment]
**Comment by [jeff-aion](https://github.com/jeff-aion)** (on Tuesday Aug 28, 2018 at 20:41 GMT)

This depends on #207, since the generated code will call these specialized constructors.
